spec/unit/classes/server/config_spec.rb

```diff
@@ -36,15 +39,112 @@
         .that_requires('Package[policycoreutils-python-utils]')
     end
 
-    it 'has the correct systemd-override file' do
+    it 'removes the old systemd-override file' do
+      is_expected.to contain_file('old-systemd-override')
+        .with(ensure: 'absent', path: '/etc/systemd/system/postgresql.service')
+    end
+
+    it 'has the correct systemd-override drop file' do
       is_expected.to contain_file('systemd-override').with(
-        ensure: 'file', path: '/etc/systemd/system/postgresql.service',
+        ensure: 'file', path: '/etc/systemd/system/postgresql.service.d/postgresql.conf',
         owner: 'root', group: 'root'
-      )
+      ) .that_requires('File[systemd-conf-dir]')
     end
-    it 'has the correct systemd-override file #content' do
+
+    it 'has the correct systemd-override file #regex' do
       is_expected.to contain_file('systemd-override') \
-        .with_content(%r{.include \/usr\/lib\/systemd\/system\/postgresql.service})
+        .with_content(%r{(?!^.include)})
+    end
+
+    #context 'RHEL 7 host with Puppet 5' do
+    #  let(:facts) {
+    #    facts.merge({ 'puppetversion' => '5.0.2' })
+    #  }
+
+      it 'has systemctl restart command' do
+        is_expected.to contain_exec('restart-systemd').with({
+          :command     => 'systemctl daemon-reload',
+          :refreshonly => true,
+          :path        => '/bin:/usr/bin:/usr/local/bin',
+        })
+      end
+    #end
+
+   describe 'with manage_package_repo => true and a version' do
+      let(:pre_condition) do
+        <<-EOS
+          class { 'postgresql::globals':
+            manage_package_repo => true,
+            version => '9.4',
+          }->
+          class { 'postgresql::server': }
+        EOS
+      end
+
+      it 'has the correct systemd-override file' do
+        is_expected.to contain_file('systemd-override').with(
+          ensure: 'file', path: '/etc/systemd/system/postgresql-9.4.service.d/postgresql-9.4.conf',
+          owner: 'root', group: 'root'
+        )
+      end
+
+      it 'has the correct systemd-override file #regex' do
+        is_expected.to contain_file('systemd-override') \
+          .without_content(%r{\.include})
+      end
+    end
+  end
+
+  describe 'on Redhat 8' do
+    let(:facts) do
+      {
+        concat_basedir: tmpfilename('server'),
```

Comment:
[Concat basedir is dead](https://github.com/puppetlabs/puppetlabs-postgresql/commit/3fba56bd8ab36e4862f9ea7b2e9cc69206271531)
```suggestion
```